File: src/test/java/com/automation/um/pages/components/UserFormModal.java
package com.automation.um.pages.components;

import com.automation.um.pages.BasePage;
import java.util.Map;
import org.openqa.selenium.WebDriver;

public class UserFormModal extends BasePage {
  public UserFormModal(WebDriver driver) {
    super(driver);
  }

  public void fill(Map<String, String> user) {
    if (user.containsKey("firstName")) fillInput("user-firstName", user.get("firstName"));
    if (user.containsKey("lastName")) fillInput("user-lastName", user.get("lastName"));
    if (user.containsKey("email")) fillInput("user-email", user.get("email"));
    if (user.containsKey("phone")) fillInput("user-phone", user.get("phone"));
    if (user.containsKey("role")) selectOptionByValue("user-role", user.get("role"));
    if (user.containsKey("status")) selectOptionByValue("user-status", user.get("status"));
  }

  public void submit() {
    el("user-form-submit").click();
  }

  public void cancel() {
    el("user-form-cancel").click();
  }

  public boolean hasValidationError(String text) {
    return driver.getPageSource().contains(text);
  }

  public boolean hasServerError(String text) {
    return driver.getPageSource().contains(text);
  }

  public boolean isOpen() {
    return isVisible("user-form-submit");
  }
}

File: playwright/pages/components/UserFormModal.js
import { BasePage } from "../BasePage.js";

export class UserFormModal extends BasePage {
  /**
   * @param {Record<string, string>} user
   */
  async fill(user) {
    if (user.firstName !== undefined) await this.fillInput("user-firstName", user.firstName);
    if (user.lastName !== undefined) await this.fillInput("user-lastName", user.lastName);
    if (user.email !== undefined) await this.fillInput("user-email", user.email);
    if (user.phone !== undefined) await this.fillInput("user-phone", user.phone);
    if (user.role !== undefined) await this.selectOptionByValue("user-role", user.role);
    if (user.status !== undefined) await this.selectOptionByValue("user-status", user.status);
  }

  async submit() {
    await this.testId("user-form-submit").click();
  }

  async cancel() {
    await this.testId("user-form-cancel").click();
  }

  async hasValidationError(text) {
    const content = await this.page.content();
    return content.includes(text);
  }

  async hasServerError(text) {
    const content = await this.page.content();
    return content.includes(text);
  }

  async isOpen() {
    return this.isVisible("user-form-submit");
  }
}
